core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -100,6 +101,7 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       pollExpected = true
       poll()
     case RequestMessages(topics) =>
+      context.watch(sender())
```

Comment:
>  I am pretty sure that there is relatively small number of sources which uses the same connection

If that is the case then watch is enough, coupled with handling the Terminated case. It is much better to use watch instead of a hand-rolled alternative. So let's keep it as it is for now.